client/app/services/group/currentGroup.service.js

```diff
@@ -9,18 +9,39 @@
 */
 export default class CurrentGroup {
 
-  constructor() {
+  constructor(User, Authentication) {
+    "ngInject";
     Object.assign(this, {
-      value: {}
+      value: {},
+      User,
+      Authentication
     });
   }
 
   set(value) {
     angular.copy(value, this.value);
+    if (angular.isUndefined(this.Authentication.data)) {
+      this.Authentication.update();
```

Comment:
I wasn't sure if I should put this check since it's handled during the routing, and was hoping you'd comment on this. Thanks, removed it in the new commit.